File: 2571-find-the-pivot-integer/find-the-pivot-integer.java
class Solution {
    public int pivotInteger(int n) {
        int right_side = ((n*n)+n) / 2;

        int x = (int) Math.sqrt(right_side);
        if(x*x == right_side)
        {
            return x;
        }
        return -1;
        
    }
}

File: 2571-find-the-pivot-integer/find-the-pivot-integer.js
/**
 * @param {number} n
 * @return {number}
 */
var pivotInteger = function(n) {

    let right_side = ((n*n)+n) / 2;
    let x =Math.floor( Math.sqrt(right_side));

    if(x*x === right_side)
    {
        return x;
    }
    return -1;
};